inc/sitemaps/class-sitemap-cache-data.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Sitemap Cache Data object, manages sitemap data stored in cache
+ */
+class WPSEO_Sitemap_Cache_Data implements WPSEO_Sitemap_Cache_Data_Interface, Serializable {
+
+	/** @var string Sitemap XML data. */
+	private $sitemap = '';
+
+	/** @var string Status of the sitemap, usable or not. */
+	private $status = self::UNKNOWN;
+
+	/**
+	 * Set the sitemap XML data
+	 *
+	 * @param string $sitemap XML Content of the sitemap.
+	 */
+	public function set_sitemap( $sitemap ) {
+
+		if ( ! is_string( $sitemap ) ) {
+			$sitemap = '';
+		}
+
+		$this->sitemap = $sitemap;
+
+		/**
+		 * Empty sitemap is not usable.
+		 */
+		$this->is_usable( ! empty( $sitemap ) );
```

Comment:
Shouldn't `is_usable()` be responsible for checking whether or not the sitemap is empty or are there situations where it's fine if it's empty?